inc/form.class.php

```diff
@@ -2028,34 +2028,57 @@ public static function import(PluginFormcreatorImportLinker $importLinker, $form
       set_time_limit(30);
 
       $form_obj = new self;
-      $entity   = new Entity;
-      $form_cat = new PluginFormcreatorCategory;
+
+      $forms_id = plugin_formcreator_getFromDBByField(
+         $form_obj,
+         'uuid',
+         $form['uuid']
+      );
 
       // retrieve foreign keys
-      if (!isset($form['_entity'])
-          || !$form['entities_id']
-             = plugin_formcreator_getFromDBByField($entity,
-                                                   'completename',
-                                                   $form['_entity'])) {
-         $form['entities_id'] = $_SESSION['glpiactive_entity'];
+      $entity = new Entity();
+      $entityFk = Entity::getForeignKeyField();
+      $entityId = $_SESSION['glpiactive_entity'];
+      if (isset($form['_entity'])) {
+         plugin_formcreator_getFromDBByField(
+            $entity,
+            'completename',
+            $form['_entity']
+         );
+         // Check rights on the destination entity of the form
+         if (!$entity->isNewItem() && $entity->canUpdateItem()) {
+            $entityId = $entity->getID();
+         } else {
+            if ($forms_id !== false) {
+               // The form is in an entity where we don't have UPDATE right
+               Session::addMessageAfterRedirect(
+                  sprintf(__('The form %1$s already exists and is in an unmodifiable entity.', 'formcreator'), $form['name']),
+                  false,
+                  WARNING
+               );
+               return false;
+            }
+         }
       }
-      if (!isset($form['_plugin_formcreator_categories_id'])
-          || !$form['_plugin_formcreator_categories_id']
-             = plugin_formcreator_getFromDBByField($form_cat,
-                                                   'completename',
-                                                   $form['_plugin_formcreator_category'])) {
-         $form['plugin_formcreator_categories_id'] = 0;
+      $form[$entityFk] = $entityId;
+
+      $formCategory = new PluginFormcreatorCategory();
+      $formCategoryFk = PluginFormcreatorCategory::getForeignKeyField();
+      $formCategoryId = 0;
+      if (isset($form['_plugin_formcreator_category'])) {
+         $formCategoryId = $formCategory->import([
+            'completename' => $form['_plugin_formcreator_category'],
```

Comment:
plugin_formcreator_getFromDBByField will get the category if it exists. If not we need to create it,  then let's use CommonTreeDropdown::import() which handles both operations.